File: frontend/dating-website/src/app/models/UserProfile.ts
import { Regions } from '../enums/regions';

export interface UserProfile{
    id: string | undefined
    userName: string
    firstName: string
    lastName: string
    birthDate: string
    gender: Boolean
    email: string
    phoneNumber: string
    region: Regions
    town: string
    photo: string
}

File: backend/api/Dating.Logic/Models/UserProfile.cs
using Microsoft.AspNetCore.Identity;
using System;
using System.ComponentModel.DataAnnotations;

namespace Dating.Logic.Models
{
    public class UserProfile
    {
        public Guid Id { get; set; }

        [Required]
        [MaxLength(450)]
        public string AspNetUserId { get; set; }

        public IdentityUser AspNetUser { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public DateTime BirthDate { get; set; }

        public bool Gender { get; set; }

        public string PhoneNumber { get; set; }

        public string Region { get; set; }

        public string Town { get; set; }
    }
}
